spyder/api/plugins.py

```diff
@@ -1607,14 +1607,10 @@ def __init__(self, parent, configuration):
         widget.set_icon(self.get_icon())
         widget.set_name(self.NAME)
 
-        # TODO: Streamline this by moving to postvisible setup
         # Render all toolbars as a final separate step on the main window
         # in case some plugins want to extend a toolbar. Since the rendering
         # can only be done once!
-        widget._main_toolbar._render()
-        widget._corner_toolbar._render()
-        for __, toolbar in widget._auxiliary_toolbars.items():
-            toolbar._render()
+        widget.render_toolbars()
```

Comment:
I added get methods as suggested but added this extra method to encapsulate all.